### ISSUE TITLE
Collecting custom data for registration event

### ISSUE THREAD

**Issue by yuha0** [sensu/sensu#1237]
I found the client registration feature in the recent version pretty useful. However, the information in the `event` is not enough for me. 

I am in a situation that we need to know client's IP, memory/CPU usage... etc. Is it possible to attach some custom data (by collecting `STDOUT` from running a script just like executing a `check`, for example) when `client` creating a registration `event`?

**Comment by calebhailey** [comment]
@yuha0 the registration event will include [custom attributes][1] defined in the [client definition][2] (e.g. `/etc/sensu/conf.d/client.json`). It should be pretty easy to add custom attributes to the client definition via configuration management (or other automation tooling) when a new node is initially provisioned (e.g. ip address, cpu count, memory capacity, etc). In fact, the [example event data shown here][3] shows some custom client definition attributes, including an IP address. 

I hope this helps. 😄 

[1]: https://sensuapp.org/docs/latest/clients#custom-attributes
[2]: https://sensuapp.org/docs/latest/clients#client-configuration 
[3]: https://sensuapp.org/docs/latest/events#event-data

**Comment by yuha0** [comment]
Hi @calebhailey Thanks for the link. I know I can add hard-coded attributes. But in the event, what if I need to send some information that will be available only after starting sensu-client?

Let's say I have a machine installed sensu-client and is configured to talk to a remote RabbitMQ. When it's booting up for the first time, I want it to check its local IP (for example, run `ifconfig`), which is assigned by a router, and include the IP address in the registration event. How can I do that?

**Comment by calebhailey** [comment]
@yuha0: this could easily be provided via a configuration management solution (e.g. Chef, Puppet, or Ansible) or other automation tooling, which could wait until an IP address is available, then write the client configuration and start the Sensu client. This is a pretty common use case for Sensu users.

**Comment by yuha0** [comment]
@calebhailey I've considered this approach but in my current project, sensu-client may be the only service we have control on the client machines. I think I will go with a normal check if there's no easy way to do it with registration event.

**Comment by calebhailey** [comment]
@yuha0 yeah, simple checks to monitor these resources should also do the trick. The registration event handler is useful for "one time registration" scenarios, which registration can include custom metadata, but it doesn't sound like that's your issue. Monitoring system resources should be done with checks.

PS - if you need to use a [subscription check][1] to monitor system resources that may vary from system to system, please consider the use of [check command tokens][2]. 

I'm going to go ahead and close this issue now. Please feel free to create a new issue if you should have any further questions. 

[1]: https://sensuapp.org/docs/latest/checks#subscription-checks
[2]: https://sensuapp.org/docs/latest/checks#check-command-tokens
